src/main/java/org/terasology/alterationEffects/regenerate/RegenerationAlterationEffect.java

```diff
@@ -1,138 +1,74 @@
-     * @param duration      The duration of the regen effect.
-     */
     @Override
-    public void applyEffect(EntityRef instigator, EntityRef entity, float magnitude, long duration) {
-        // First, determine if the entity already has a regeneration component attached. If so, just replace the amount
-        // and last regen time, and then save the component. Otherwise, create a new one and attach it to the entity.
-        RegenerationComponent regeneration = entity.getComponent(RegenerationComponent.class);
-        if (regeneration == null) {
-            regeneration = new RegenerationComponent();
-            regeneration.regenerationAmount = TeraMath.floorToInt(magnitude);
-            regeneration.lastRegenerationTime = time.getGameTimeInMs();
-            entity.addComponent(regeneration);
-        } else {
-            regeneration.regenerationAmount = TeraMath.floorToInt(magnitude);
-            regeneration.lastRegenerationTime = time.getGameTimeInMs();
-            entity.saveComponent(regeneration);
-        }
-
-        // Send out this event to collect all the duration and magnitude modifiers and multipliers that can affect this
-        // regeneration effect.
-        OnEffectModifyEvent effectModifyEvent = entity.send(new OnEffectModifyEvent(instigator, entity, 0, 0, this, ""));
-        long modifiedDuration = 0;      // This will keep track of the current modified duration.
-        boolean modifiersFound = false; // This flag will keep track if there were any modifiers collected in the event.
-
-        // If the effect modify event is consumed, don't apply this health regeneration effect.
-        if (!effectModifyEvent.isConsumed()) {
-            /*
-            Get the magnitude result value and the shortest duration, and assign them to the modifiedMagnitude and
-            modifiedDuration respectively.
-
-            The shortest duration is used as the effect modifier associated with that will expire in the shortest
-            amount of time, meaning that this effect's total magnitude and next remaining duration will have to be
-            recalculated.
-            */
-            float modifiedMagnitude = effectModifyEvent.getMagnitudeResultValue();
-            modifiedDuration = effectModifyEvent.getShortestDuration();
-
-            // If there's at least one duration and magnitude modifier, set the effect's magnitude and the modifiersFound flag.
-            if (!effectModifyEvent.getDurationModifiers().isEmpty() && !effectModifyEvent.getMagnitudeModifiers().isEmpty()) {
-                regeneration.regenerationAmount = (int) modifiedMagnitude;
-                modifiersFound = true;
-            }
-        }
-
-        // Save the component so the latest changes to it don't get lost when the game's exited.
-        entity.saveComponent(regeneration);
+    protected RegenerationComponent upsertComponent(Optional<RegenerationComponent> maybeComponent,
```

Comment:
Neither `upsertComponent` nor `updateComponent` are ever called for this effect since the default implementation of the component-based effect is overweritten below.

I'm not sure whether this actually provides any value, but it is the direct translation of the code that was previously executed when applying the effect without `id`. :shrug: